feat: use OTLP gRPC exporter by default (#319)
signalfx/splunk-otel-js: examples/express/README.md

## Setup

> Before starting install dependencies and run the collector for the examples which require it:

Reviewer: ```suggestion
Before you start, install the dependencies and run the collector for the examples that require it:
```

    - `@opentelemetry/instrumentation-http` and
    - `@splunk/otel`.
2. Configuration for the SDK.
3. SDK initialization before running the application.

> All those steps are done for you in this example:
> 
> 1. Additional packages are already specified in the `package.json`.
> 2. All the configuration is included in [an environment file `.env.collector`](./.env.collector).
> 3. SDK intitialization in [tracer.js](./tracer.js).
> 
> All that's left is to run it:

Reviewer: ```suggestion
All the following steps are done for you in the example:

1. Additional packages are already specified in the `package.json` file.
2. All the configuration is included in [an environment file, `.env.collector`](./.env.collector).
3. SDK initialization happens in [tracer.js](./tracer.js).

All that's left is to run it:
```

```shell
npm run server:collector
# In a separate terminal:
npm run client:collector

```

See the exact commands in [package.json](./package.json).

### Instrumented via OTel SDK, Jaeger Exporter

> There's an [tracer setup](./tracer.jaeger.js) to showcase replacing the default OTLP/gRPC Exporter with one that exports in Jaeger format. The configuration is similar to the plain OTel SDK setup, but the Exporter(`OTEL_TRACES_EXPORTER`) has to be replaced by the default Jaeger Exporter which is bundled with the SDK: `jaeger-thrift-http`.

Reviewer: ```suggestion
The [tracer setup file](./tracer.jaeger.js) showcases how to replace the default OTLP/gRPC Exporter with a Jaeger exporter. The configuration is similar to the default OTel SDK setup, but the exporter (`OTEL_TRACES_EXPORTER`) has to be replaced by the default Jaeger Exporter which is bundled with the SDK: `jaeger-thrift-http`.
```

That's it! To run the example with the Jaeger Exporter:

```shell
npm run server:jaeger
# In a separate terminal:
npm run client:jaeger
```

### Instrumented via OTel SDK, exporting to Splunk APM using Jaeger Exporter

> Instead of using OTel Collector to forward the telemetry data to Splunk APM, one can also send it directly from the application.
> Once `.env` file is created(see file [.env.jaeger-splunk](./.env.jaeger-splunk) for an example) and the access token is substituted into the file. Run it using that:

Reviewer: ```suggestion
Instead of using the OTel Collector to forward telemetry data to Splunk APM, you can also send it directly from the application. Once the `.env` file is created (see [.env.jaeger-splunk](./.env.jaeger-splunk) for an example) and the access token is replaced in the file, run the collector using the following commands:
```

```shell
npm run server
# In a separate terminal:
npm run client
```

> If unsure about the values to use for `.env` but familiar with configuring OpenTracing SDK, one can alternatively also use [.env.opentracing](./.env.opentracing) as the basis - the OTel configuration will be automatically derived from that in this example(see [utils.js](./utils.js) for the conversions).

Reviewer: ```suggestion
If are not sure about the values to use for `.env`, but you're familiar with configuring the OpenTracing SDK, you can also use [.env.opentracing](./.env.opentracing) as the baseline configuration: The OTel configuration is automatically derived from the OpenTracing settings in the example (see [utils.js](./utils.js) for the conversions).
```

Reviewer: @Rauno56 This is pretty cool. Could users use this method when migrating?

Author: Not... directly. The example also logs the configuration out so they will get an idea of the resulting configuration, but there's no "automatic" way implemented.

They could use the function from `utils.js` themselves though, but I think just reading the code will be informative enough for them to do the conversion once and for all.

### Instrumented via legacy OpenTracing SDK

For comparison this example also includes the legacy OpenTracing SDK setup which can be run using predefined commands:

```shell
npm run server:opentracing
# In a separate terminal:
npm run client:opentracing
```

> These commands require [SignalFX Tracing Library](https://github.com/signalfx/signalfx-nodejs-tracing)-compatible configuration in the `.env`(see the [example](./.env.opentracing)).

Reviewer: ```suggestion
These commands require a configuration in the `.env` file compatible with the [SignalFX Tracing Library](https://github.com/signalfx/signalfx-nodejs-tracing). See the [example](./.env.opentracing).
```
